shell/platform/android/test/io/flutter/embedding/engine/systemchannels/SettingsChannelTest.java

```diff
@@ -0,0 +1,70 @@
+package io.flutter.embedding.engine.systemchannels;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import android.annotation.TargetApi;
+import android.util.DisplayMetrics;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.BasicMessageChannel;
+import java.nio.ByteBuffer;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.robolectric.annotation.Config;
+
+@Config(manifest = Config.NONE)
+@RunWith(AndroidJUnit4.class)
+public class SettingsChannelTest {
+  @Test
+  @TargetApi(33)
+  @Config(sdk = 33)
+  public void setDisplayMetricsDoesNothingOnAPILevel33() {
+    final DartExecutor executor = mock(DartExecutor.class);
+    executor.onAttachedToJNI();
+    final SettingsChannel settingsChannel = new SettingsChannel(executor);
+
+    final ArgumentCaptor<ByteBuffer> messageCaptor = ArgumentCaptor.forClass(ByteBuffer.class);
+
+    settingsChannel.startMessage().setDisplayMetrics(mock(DisplayMetrics.class)).send();
+
+    verify(executor).send(eq("flutter/settings"), messageCaptor.capture(), isNull());
+  }
+
```

Comment:
Can you add a test with minSdk = 33 that will eventually exercise the api 34 code or at minimum a todo with link to a bug in this file to add a tests once api 34 is available against roboletric.

Reply:
Added a TODO that points to https://github.com/flutter/flutter/issues/128825